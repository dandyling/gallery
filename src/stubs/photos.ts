import { Basic } from "unsplash-js/dist/methods/photos/types";

export const stubbedPhotos: RecursivePartial<Basic[]> = [
  {
    id: "JgOeRuGD_Y1",
    alt_description: "aerial photo of brown moutains",
    urls: {
      raw: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1",
      full: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=srgb&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=200",
    },
    user: {
      id: "k1Rq8ji2l24",
      username: "heytowner",
      name: "JOHN TOWNER",
      first_name: "JOHN",
      last_name: "TOWNER",
      twitter_username: null,
      instagram_username: "heytowner",
    },
  },
  {
    id: "JgOeRuGD_Y2",
    alt_description: "aerial photo of brown moutains",
    urls: {
      raw: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1",
      full: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=srgb&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=200",
    },
    user: {
      id: "k1Rq8ji2l24",
      username: "heytowner",
      name: "JOHN TOWNER",
      first_name: "JOHN",
      last_name: "TOWNER",
      twitter_username: null,
      instagram_username: "heytowner",
    },
  },
  {
    id: "JgOeRuGD_Y3",
    alt_description: "aerial photo of brown moutains",
    urls: {
      raw: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1",
      full: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=srgb&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=200",
    },
    user: {
      id: "k1Rq8ji2l24",
      username: "heytowner",
      name: "JOHN TOWNER",
      first_name: "JOHN",
      last_name: "TOWNER",
      twitter_username: null,
      instagram_username: "heytowner",
    },
  },
  {
    id: "JgOeRuGD_Y4",
    alt_description: "aerial photo of brown moutains",
    urls: {
      raw: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1",
      full: "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=srgb&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1477346611705-65d1883cee1e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTA1MHwwfDF8c2VhcmNofDF8fFdhbGxwYXBlcnN8ZW58MHwwfHx8MTYyNTAzNDM0MQ&ixlib=rb-1.2.1&q=80&w=200",
    },
    user: {
      id: "k1Rq8ji2l24",
      username: "heytowner",
      name: "JOHN TOWNER",
      first_name: "JOHN",
      last_name: "TOWNER",
      twitter_username: null,
      instagram_username: "heytowner",
    },
  },
];
