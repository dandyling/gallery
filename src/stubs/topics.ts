import { Basic } from "unsplash-js/dist/methods/topics/types";

export const stubbedTopics: RecursivePartial<Basic[]> = [
  {
    id: "J9yrPaHXRQ1",
    title: "Technology",
    description:
      '<p>As technology is evolving quickly, what was relevant one year ago may not be today. This is why we’ve teamed up with Dell XPS to build an ever evolving archive of technology. From photos of the latest trends in tech to every day shots of your beloved laptops and mobile phones — submit your photos to the Technology Topic for a chance to be featured. </p>\r\n\r\n<p>Curated by Dell XPS. Learn more about this topic takeover <a href="/blog/showcasing-the-best-of-technology-with-dell-xps/">here</a>.</p>',
    cover_photo: {
      id: "-K6JMRMj4x4",
      blur_hash: "L5PGjXt7WBRj?bxu4nM{?bxu00WB",
      description: "Flat Lay Adler Typewriter",
      alt_description:
        "aligned black and gray metal equipment part high-angle photo",
      urls: {
        raw: "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1",
        full: "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb",
        regular:
          "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max",
        small:
          "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max",
        thumb:
          "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max",
      },
    },
  },
  {
    id: "J9yrPaHXRQ2",
    title: "Technology",
    description:
      '<p>As technology is evolving quickly, what was relevant one year ago may not be today. This is why we’ve teamed up with Dell XPS to build an ever evolving archive of technology. From photos of the latest trends in tech to every day shots of your beloved laptops and mobile phones — submit your photos to the Technology Topic for a chance to be featured. </p>\r\n\r\n<p>Curated by Dell XPS. Learn more about this topic takeover <a href="/blog/showcasing-the-best-of-technology-with-dell-xps/">here</a>.</p>',
    cover_photo: {
      id: "-K6JMRMj4x4",
      blur_hash: "L5PGjXt7WBRj?bxu4nM{?bxu00WB",
      description: "Flat Lay Adler Typewriter",
      alt_description:
        "aligned black and gray metal equipment part high-angle photo",
      user: {
        id: "ZbAEdl9dx6k",
        username: "florianklauer",
        name: "Florian Klauer",
        first_name: "Florian",
        last_name: "Klauer",
        twitter_username: null,
        instagram_username: "florianklauer",
      },
    },
  },
  {
    id: "J9yrPaHXRQ3",
    description:
      '<p>As technology is evolving quickly, what was relevant one year ago may not be today. This is why we’ve teamed up with Dell XPS to build an ever evolving archive of technology. From photos of the latest trends in tech to every day shots of your beloved laptops and mobile phones — submit your photos to the Technology Topic for a chance to be featured. </p>\r\n\r\n<p>Curated by Dell XPS. Learn more about this topic takeover <a href="/blog/showcasing-the-best-of-technology-with-dell-xps/">here</a>.</p>',
    cover_photo: {
      id: "-K6JMRMj4x4",
      blur_hash: "L5PGjXt7WBRj?bxu4nM{?bxu00WB",
      description: "Flat Lay Adler Typewriter",
      alt_description:
        "aligned black and gray metal equipment part high-angle photo",
      urls: {
        raw: "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1",
        full: "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb",
        regular:
          "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max",
        small:
          "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max",
        thumb:
          "https://images.unsplash.com/27/type-set.jpg?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max",
      },
      user: {
        id: "ZbAEdl9dx6k",
        username: "florianklauer",
        name: "Florian Klauer",
        first_name: "Florian",
        last_name: "Klauer",
        twitter_username: null,
        instagram_username: "florianklauer",
      },
    },
  },
];
