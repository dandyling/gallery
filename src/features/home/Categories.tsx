import { Flex, FlexProps, Heading, Image, Text } from "@chakra-ui/react";
import styled from "@emotion/styled";
import React from "react";
import { Basic } from "unsplash-js/dist/methods/topics/types";

interface CategoriesProps extends FlexProps {
  topics: Basic[];
  onClickCategory(category: string): void;
}

export const Categories = (props: CategoriesProps) => {
  const { onClickCategory, topics, ...rest } = props;

  return (
    <Flex direction="column" {...rest}>
      <Heading as="h2" fontSize="xl" py="2">
        Categories
      </Heading>
      <Flex overflowX="scroll">
        {topics.map((topic, i) => {
          const { cover_photo, title, id } = topic;

          const handleClick = () => {
            return onClickCategory(title);
          };

          return (
            <RatioContainer
              mr="2"
              minHeight="35vw"
              maxHeight="30ch"
              onClick={handleClick}
              key={`${i} - ${id}`}
              justifyContent="center"
            >
              <Image
                borderRadius="2xl"
                objectFit="cover"
                minHeight="100%"
                src={cover_photo?.urls.thumb}
                alt={
                  cover_photo?.alt_description ??
                  `Photo by ${cover_photo?.user}`
                }
              />
              <Text
                color="white"
                textAlign="center"
                position="absolute"
                bottom="2"
                fontWeight="bold"
                shadow="lg"
                lineHeight="1.1"
              >
                {title}
              </Text>
            </RatioContainer>
          );
        })}
      </Flex>
    </Flex>
  );
};

interface RatioContainerProps extends FlexProps {
  aspectRatio?: string;
}

const RatioContainer = styled(Flex)<RatioContainerProps>`
  aspect-ratio: ${(props) => props.aspectRatio ?? "3 / 4"};
  position: relative;
`;
