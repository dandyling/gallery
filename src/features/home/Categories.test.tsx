import { render, screen } from "@testing-library/react";
import { shallow } from "enzyme";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { Basic } from "unsplash-js/dist/methods/topics/types";
import { stubbedTopics } from "../../stubs/topics";
import { Categories } from "./Categories";

test("renders the Category header", () => {
  render(
    <BrowserRouter>
      <Categories
        topics={stubbedTopics as Basic[]}
        onClickCategory={() => {}}
      />
    </BrowserRouter>
  );
  const linkElement = screen.getByText(/Categories/i);
  expect(linkElement).toBeInTheDocument();
});

test("renders the correct number of cover images", () => {
  const { container } = render(
    <BrowserRouter>
      <Categories
        topics={stubbedTopics as Basic[]}
        onClickCategory={() => {}}
      />
    </BrowserRouter>
  );
  const images = container.getElementsByTagName("img");
  expect(images.length).toEqual(3);
});

test("render header if title is passed in", () => {
  const { container } = render(
    <BrowserRouter>
      <Categories
        topics={stubbedTopics.slice(0, 1) as Basic[]}
        onClickCategory={() => {}}
      />
    </BrowserRouter>
  );
  const h3 = container.getElementsByTagName("h3");
  expect(h3.length).toEqual(1);
});

test("doesn't render header if title is not passed in", () => {
  const { container } = render(
    <BrowserRouter>
      <Categories
        topics={stubbedTopics.slice(2) as Basic[]}
        onClickCategory={() => {}}
      />
    </BrowserRouter>
  );
  const h3 = container.getElementsByTagName("h3");
  expect(h3.length).toEqual(0);
});

test("renders correctly", () => {
  const wrapper = shallow(
    <BrowserRouter>
      <Categories
        topics={stubbedTopics as Basic[]}
        onClickCategory={() => {}}
      />
    </BrowserRouter>
  );
  expect(wrapper).toMatchSnapshot();
});
