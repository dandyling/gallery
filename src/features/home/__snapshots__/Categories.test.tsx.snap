// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders the Category header 1`] = `ShallowWrapper {}`;
